skimage/measure/profile.py

```diff
@@ -109,3 +124,229 @@ def _line_profile_coordinates(src, dst, linewidth=1):
+        since the line is 3d, this value is the diameter
+        of a 3d cylinder following the scan line.
+
+    Returns
+    -------
+    perp_array : array, shape (3, M, N), float
+        The coordinates of the profile along the scan line. The length of the
+        profile, M, is the ceil of the computed length of the scan line.
+        The line width, N, is the number of sample points taken at each unit of the scan line.
+
+    Notes
+    -----
+    This is a utility method meant to be used internally by skimage functions.
+    The destination point is included in the profile, in contrast to
+    standard numpy indexing.
+    """
+    src_pln, src_row, src_col = src = np.asarray(src, dtype=float)
+    dst_pln, dst_row, dst_col = dst = np.asarray(dst, dtype=float)
+    d_pln, d_row, d_col = dst - src
+
+    # Get one unit vector perpendicular to direction vector to find a point
+    # that is one unit distance away from the destination vector
+    # (ex: ix + jy + kz = 0, then we can use x = y = 1)
+    # Try with z first if it is not 0, then the same for x, otherwise pick y
+    # We subtract 1 from linewidth to change from pixel-counting
+    # (make this line 3 pixels wide) to point distances (the
+    # distance between pixel centers)
+    if d_pln != 0:
+        # try finding the solution to ix + jy + kz = 0 for x = 1 and y = 1
+        dim_z = - (d_row + d_col) / d_pln
+        length_vector = np.sqrt(2 + dim_z ** 2)
+        col_width = row_width = (linewidth - 1) / (2 * length_vector)
+        slice_width = (linewidth - 1) * (dim_z / 2 * length_vector)
+    elif d_row != 0:
+        # try finding the solution to ix + jy + kz = 0 for y = 1 and z = 1
+        dim_x = - (d_pln + d_col) / d_row
+        length_vector = np.sqrt(2 + dim_x ** 2)
+        col_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        row_width = (linewidth - 1) * (dim_x / length_vector) / 2
+    else:
+        # try finding the solution to ix + jy + kz = 0 for x = 1 and z = 1
+        dim_y = - (d_row + d_pln) / d_col
+        length_vector = np.sqrt(2 + dim_y ** 2)
+        row_width = slice_width = (linewidth - 1) / (2 * length_vector)
+        col_width = (linewidth - 1) * (dim_y / length_vector) / 2
+
+    # we add one above because we include the last point in the profile
+    # (in contrast to standard numpy indexing)
+    length = np.ceil(np.linalg.norm([d_pln, d_row, d_col]) + 1)
+
+    line_col = np.linspace(src_col, dst_col, length)
+    line_row = np.linspace(src_row, dst_row, length)
+    line_pln = np.linspace(src_pln, dst_pln, length)
+
+    # find divisor to get only first half of array and center point if odd
+    first_half_index = np.int(np.ceil(np.float(linewidth)/2))
+    if first_half_index < 1:
+        first_half_index = 1
+
+    perp_rows = np.array([np.linspace(row_i - row_width, row_i + row_width,
+                                      linewidth)[:first_half_index] for row_i in line_row])
```

Comment:
Actually, instead of doing the indexing trick with `first_half_index` at all, you could do:

```
perp_rows = [np.linspace(row_i - row_width, row_i, np.ceil(linewidth / 2))
             for row_i in line_row]
```

Note also that you don't need the call to `np.array`. A single call, below with `perp_array`, will take care of all the nested lists.